src/core/unit.c

```diff
@@ -3830,6 +3830,10 @@ void unit_reset_failed(Unit *u) {
 
         ratelimit_reset(&u->start_ratelimit);
         u->start_limit_hit = false;
+        u->debug_invocation = false;
+        ExecContext *ec = unit_get_exec_context(u);
+        if (ec)
+                unit_overwrite_log_level_max(u, ec->log_level_max);
```

Comment:
Why not call service_set_debug_invocation helper here?

Reply:
because this is a unit, that's for the service, so we need to actually check that there is an EC

Reply:
please just move this into service_reset_failed()?

Reply:
ok